From 37c7a798719f1d04326d36c35711c4249bc7492e Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Sun, 12 May 2024 14:32:20 +0200
Subject: [PATCH] Correctly use PTR_ERR in pcie_bf_mimo_ctrl_decode

Correctly use PTR_ERR instead of cast to unsigned int in
pcie_bf_mimo_ctrl_decode if fp_data pointer contains errors.

This fix a compilation warning on 64Bit:

1325 | filename, (unsigned int)fp_data);
| ^

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 hif/pcie/pcie.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/hif/pcie/pcie.c
+++ b/hif/pcie/pcie.c
@@ -1449,8 +1449,8 @@ static void pcie_bf_mimo_ctrl_decode(str
 			       &fp_data->f_pos);
 		filp_close(fp_data, current->files);
 	} else {
-		wiphy_err(priv->hw->wiphy, "Error opening %s! %x\n",
-			  filename, (unsigned int)fp_data);
+		wiphy_err(priv->hw->wiphy, "Error opening %s! %ld\n",
+			  filename, PTR_ERR(fp_data));
 	}
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(5,10,0)
